### PR TITLE
[expression.dd] Improve formatting for comparison operators

### PR DESCRIPTION
**Pull request by ntrel**
* Fix closing `<p>` tags before `<div>`, lists or code examples to prevent empty lines in HTML.
* Add *Aggregate Equality* subheading under *Equality Expressions* & move paragraph about arrays above it.
* Add *Array Comparisons* subheading under *Relational Expressions*.
* Improve example showing runtime null object equality error.
* Move paragraph about `o1.opCmp(o2)` under existing *Class Comparisons* subheading.

### CODE REVIEW

File: spec/expression.dd [hunk 1]
```diff
@@ -507,18 +507,25 @@ $(GNAME EqualExpression):
         Otherwise, the bit patterns of the common type are compared for equality.
     )
 
-    $(P For complex numbers, equality is defined as equivalent to:
+    $(P For complex numbers, equality is defined as equivalent to:)
```
Comment (maxhaton) [inline spec/expression.dd:511]: We have deprecated built-in complex numbers, perhaps this sentence should be reworded or removed?

Comment (ntrel) [reply spec/expression.dd]: I've indicated this now.

File: spec/expression.dd [hunk 2]
```diff
@@ -677,13 +670,24 @@ $(H3 $(LEGACY_LNAME2 class_comparisons, class-comparisons, Class comparisons))
...
+        a $(CODE null) class reference is invalid, as $(CODE null) has no contents.)
 
+        $(SPEC_RUNNABLE_EXAMPLE_FAIL
         ---
-        class C;
+        class C {}
         C c;
-        if (c < null)  // error
-            ...
+        if (c < null) {}  // compile-time error
```
Comment (dkorpel) [inline spec/expression.dd:679]: Shouldn't this be put in a `void main() {}`? A top-level if-statement doesn't parse.

**Review by dkorpel — APPROVED** [review]

File: spec/expression.dd (hunk 2 above)
Comment (ntrel) [reply spec/expression.dd]: Examples that don't contain `main` are wrapped in a `main` function (and have `stdio` imported). So C here is a nested class.

Comment (dkorpel) [reply spec/expression.dd]: Good to know

**Pull request merged**